src/goto-instrument/contracts/dynamic-frames/dfcc.cpp

```diff
@@ -0,0 +1,467 @@
+
+    // contracts for function pointers must be replaced with themselves
+    // so we need to check that:
+    // - the symbol exists as a function symbol
+    // - the symbol exists as a pure contract symbol
+    // - the function symbol is not already swapped for contract checking
+    // - the function symbol is not already swapped with another contract for
+    // replacement
+
+    const auto str = id2string(fp_contract);
+
+    // Is it already swapped with another function for contract checking ?
+    PRECONDITION_WITH_DIAGNOSTICS(
+      to_check.find(str) == to_check.end(),
+      "Function '" + str +
+        "' used as contract for function pointer cannot be itself the object "
+        "of a contract check.");
+
+    // Is it already swapped with another function for contract checking ?
+    auto found = to_replace.find(str);
+    if(found != to_replace.end())
+    {
+      PRECONDITION_WITH_DIAGNOSTICS(
+        found->first == found->second,
+        "Function '" + str +
+          "' used as contract for function pointer already the object of a "
+          "contract replacement with '" +
+          found->second + "'");
+      log.status() << "Function pointer contract '" << fp_contract
+                   << "' already wrapped with itself in REPLACE mode"
+                   << messaget::eom;
+    }
+    else
+    {
+      // we need to swap it with itself
+      PRECONDITION_WITH_DIAGNOSTICS(
+        utils.function_symbol_exists(str),
+        "Function pointer contract '" + str + "' not found.");
+
+      PRECONDITION_WITH_DIAGNOSTICS(
+        dsl_contract_handler.pure_contract_symbol_exists(str),
+        "Contract to replace '" + str + "' not found.");
+
+      log.status() << "Wrapping function pointer contract '" << fp_contract
+                   << "' with itself in REPLACE mode" << messaget::eom;
+
+      swap_and_wrap.swap_and_wrap_replace(
+        fp_contract, fp_contract, function_pointer_contracts);
+      // remove it from the set of symbols to process
+      if(other_symbols.find(fp_contract) != other_symbols.end())
+        other_symbols.erase(fp_contract);
+    }
+  }
+
+  // instrument all other remaining functions
+  for(const auto &function_id : other_symbols)
+  {
+    // Don't instrument CPROVER and internal functions
+    if(instrument.do_not_instrument(function_id))
+    {
+      continue;
+    }
+
+    log.status() << "Instrumenting '" << function_id << "'" << messaget::eom;
+
+    instrument.instrument_function(function_id);
+  }
+
+  log.status() << "Updating goto functions" << messaget::eom;
+  goto_model.goto_functions.update();
+
+  if(!to_check.empty())
+  {
+    log.status() << "Specializing cprover_contracts functions for assigns "
+                    "clauses of at most "
+                 << max_assigns_clause_size << " targets" << messaget::eom;
+    library.specialize(max_assigns_clause_size);
+  }
+
+  log.status() << "Inhibiting front-end functions" << messaget::eom;
```

Comment:
front-end functions like `__CPROVER_object_whole` are now supposed to be mapped to their library implementation. However by security, the body of these functions get turned into `assert(false);assume(false);` so that any leftover calls in user code would make the whole analysis fail.

Reply:
Ok, but would a user be able to make sense of this status message (without your additional elaboration)?

Reply:
removed the print